npm/src/clientfileprocessing/index.ts

```diff
@@ -25,11 +25,26 @@ export class ClientFileProcessing {
 
   metadataProgressCallback(progressInformation: IProgressInformation) {
     const weightedPercent = progressInformation.percentageProcessed / 2
+
+    const fileUpload: HTMLDivElement | null =
+      document.querySelector("#file-upload")
+    const progressBar: HTMLDivElement | null =
+      document.querySelector("#progress-bar")
     const progressBarElement: HTMLDivElement | null =
       document.querySelector(".progress-display")
+    const progressLabelElement: HTMLDivElement | null =
+      document.querySelector(".progress-label")
+
+    if (fileUpload && progressBar) {
+      fileUpload.setAttribute("hidden", "true")
+      progressBar.removeAttribute("hidden")
+    }
 
-    if (progressBarElement) {
+    if (progressBarElement && progressLabelElement) {
```

Comment:
This is the same code as at line 60. It might be worth abstracting this out into a separate function:

```
renderWeightedPercent(weightedPercent: number) {
    const progressBarElement: HTMLDivElement | null =
        document.querySelector(".progress-display")
    const progressLabelElement: HTMLDivElement | null =
        document.querySelector(".progress-label")

    if (progressBarElement && progressLabelElement) {
      var stringWeightedPercentage = weightedPercent.toString()
      progressBarElement.setAttribute("value", weightedPercent.toString())
      progressLabelElement.innerText =
          "Uploading records " + stringWeightedPercentage + "%"
    }    
  }
```
Then you can calculate the weighted percent value in the calling function, and pass it into the new function